scene/gui/tabs.cpp

```diff
@@ -241,6 +241,7 @@ void Tabs::_shape(int p_tab) {
 
 	tabs.write[p_tab].xl_text = atr(tabs[p_tab].text);
 	tabs.write[p_tab].text_buf->clear();
+	tabs.write[p_tab].text_buf.instantiate();
```

Comment:
Try `tabs.write[p_tab].text_buf->set_width(-1);` instead. Recreating the whole buffer most likely is not necessary.

Or maybe try adding `_update_cache();` to the `set_tab_title`. I have not tested it, but text alignment width calculated in the `_update_cache();` is probably what's causing the issue.

Reply:
Changing the width to `-1` did it, thanks! Also, `_update_cache()` is called with every drawing update already, so that wasn't it.